src/docs/arch-btree.dox

```diff
@@ -0,0 +1,104 @@
+/*! @arch_page arch-btree B-Trees
+
+Database tables are commonly represented by a B-Tree data structure, which
+is made up of nodes that are page structures. The root and internal pages will
+only store keys and references to other pages, while leaf pages store keys,
+values and sometimes references to overflow pages. The focus here will be on
+the in-memory representation of the B-Tree which is defined in \c btmem.h. The
+on-disk representation and organization of pages is discussed in the
+@ref arch-data-file page.
+
+@section btree_btree_data_source B-Tree Data Source (WT_BTREE)
+
+As discussed in the @ref arch-dhandle page, dhandles are generic containers used
+to access various types of data sources. Dhandles that represent B-Trees contain a
+pointer to the \c WT_BTREE. The \c WT_BTREE (defined in \c btree.h) embodies both
+the on-disk and in-memory representations of the B-Tree. At a high-level, it
+contains a memory cache of key value pairs, along with functions to read and write
+data as needed to and from the data file. More specifically, the structure contains
+a reference to its corresponding generic data handle, checkpoint information, the
```

Comment:
I would use `pointer` rather than `reference`.  Later on this page you talk about `WT_REF`, that we tend to call "references".  The pointer to the generic data handle is not one of these.

(The "reference to the root page" later in this paragraph _is_ one of these.